src/xGadgetron/cGadgetron/tests/CMakeLists.txt

```diff
@@ -17,13 +17,77 @@
 #
 #=========================================================================
 
+
+function(get_zenodo_file path url md5)
+        message(STATUS "Downloading (if not already done)...
+                Path:     ${path}
+                URL:      ${url}
+                MD5:      ${md5}")
+        file(DOWNLOAD "${url}" "${path}" SHOW_PROGRESS EXPECTED_MD5 "${md5}")
+endfunction()
+
+OPTION(DOWNLOAD_ZENODO_TEST_DATA "download zenodo data for tests" ON)
+if (DOWNLOAD_ZENODO_TEST_DATA)
+        set(MR_RPE_DATA "${CMAKE_SOURCE_DIR}/data/examples/MR/zenodo/3D_RPE_Lowres.h5")
+        get_zenodo_file(
+                "${MR_RPE_DATA}"
+                "https://zenodo.org/record/4600937/files/3DRPE_Lowres.h5?download=1"
+                "a692ba3a6c456d1ad49750717a9d5dda")
+endif()
+
+OPTION(RERUN_ISMRMRD_SHEPP_LOGAN "Generate testdata compatible with ISMRMRD Version" ON)
+find_program(EXE_ISMRMRD_SL NAMES ismrmrd_generate_cartesian_shepp_logan
+                NAMES_PER_DIR
+                PATHS ${CMAKE_INSTALL_PREFIX})
+
+
+set(TESTDATA_OUTPUT_PATH "${SIRF_SOURCE_DIR}/data/examples/MR/")
+set(TESTDATA_OUTPUT_NAME_FULL "simulated_MR_2D_cartesian.h5")
+set(TESTDATA_OUTPUT_NAME_GRAPPA "simulated_MR_2D_cartesian_Grappa2.h5")
+
+set(TESTDATA_TEMP_PREFIX "testdata_tmp")
+
+if(EXE_ISMRMRD_SL AND RERUN_ISMRMRD_SHEPP_LOGAN)
+    add_custom_target(GENERATE_SIMULATED_TESTDATA ALL
+        COMMAND rm -f ${TESTDATA_OUTPUT_NAME_FULL}
+        COMMAND rm -f ${TESTDATA_OUTPUT_NAME_GRAPPA}
+        COMMAND env LD_LIBRARY_PATH=${CMAKE_INSTALL_PREFIX}/lib:$LD_LIBRARY_PATH ${EXE_ISMRMRD_SL} -r 2 -o "${TESTDATA_TEMP_PREFIX}_1.h5"
+        COMMAND env LD_LIBRARY_PATH=${CMAKE_INSTALL_PREFIX}/lib:$LD_LIBRARY_PATH ${EXE_ISMRMRD_SL} -a 2 -o "${TESTDATA_TEMP_PREFIX}_2.h5"
+        WORKING_DIRECTORY ${TESTDATA_OUTPUT_PATH})
+elseif(RERUN_ISMRMRD_SHEPP_LOGAN)
+    message(ERROR "We could not find the ISMRMRD executable for the Shepp Logan Generation." )
+else()
+    add_custom_target(GENERATE_SIMULATED_TESTDATA ALL
+        COMMAND "")
+endif()
+
 add_library(MR_TESTS_CPP_AUXILIARY ${CMAKE_CURRENT_SOURCE_DIR}/mrtest_auxiliary_funs.cpp)
 target_link_libraries(MR_TESTS_CPP_AUXILIARY PUBLIC csirf cgadgetron)
 
+add_executable(MR_PROCESS_TESTDATA ${CMAKE_CURRENT_SOURCE_DIR}/mrtests_prep_testdata.cpp)
```

Comment:
This error is because the target `PROCESS_SIMULATED_TESTDATA` can't find the gadgetron library. 
The code used to build the home `MR_PROCESS_TESTDATA` used in the target does not use any gadgetron code, but it uses our `cgadgetron` library.

Do you have an idea of what is missing? Do I need to add another dependency as 
```add_dependencies(PROCESS_SIMULATED_TESTDATA MR_PROCESS_TESTDATA cgadgetron)```?
I thought that would have been covered by linking the `PROCESS_SIMULATED_TESTDATA` publicly against cgadgetron.

Reply:
you have
```
target_link_libraries(cgadgetron PUBLIC ${GT_LIBS})
```
so that should be fine. I'm a bit confused as ` error while loading shared libraries` is normally a run-time error I thought (not a linking error). can we reproduce this somewhere else than GHA?